src/node_url.cc

```diff
@@ -407,29 +407,62 @@ int64_t ParseIPv4Number(const char* start, const char* end) {
   return strtoll(start, nullptr, R);
 }
 
+// https://url.spec.whatwg.org/#ipv4-number-parser
+bool IsIPv4NumberValid(std::string_view input) {
+  if (input.empty()) {
+    return false;
+  }
+
+  if (input.size() >= 2 && input[0] == '0') {
```

Comment:
These conditions come from the URL specs (https://url.spec.whatwg.org/#ipv4-number-parser). I can either comment a quick "why" (with what it means) or do something like:
`// ref: #ipv4-number-parser session in url specs, subtopics 4-5`
what do you think is better?

Also the link for this the session related to this function is on top of the function.

Reply:
Something like this would be nice to have:

```
A basic comment like `if a number starts with `0` it might be a number with base 8 or base 16. If not, checking if all characters are digits proves that it is a base 10 number.
```

Reply:
done!